pkg/util/util.go

```
@@ -42,7 +38,12 @@ func GetAzureStackHCIMachinesInCluster(ctx context.Context, controllerClient cli
 func RandomAlphaNumericString(n int) string {
 	result := make([]byte, n)
 	for i := range result {
-		result[i] = charSet[rnd.Intn(len(charSet))]
+		val, err := rand.Int(rand.Reader, big.NewInt(int64(len(charSet))))
+		if err != nil {
+			result[i] = charSet[i]
+		} else {
```

Can we think of an alternative here. We are not sure when we would hit this

---

We can propagate the error back to the caller and let them handle it. I can't think of any other ways to handle when we can't generate a random number.